frontend/issue-board.tsx

```diff
@@ -111,7 +114,13 @@ function IssueBoard({ issues, onUpdateIssues, onOpenDetail }: Props) {
 
   const handleChangePriority = useCallback(
     (issue: Issue, priority: Priority) => {
-      onUpdateIssues([{ id: issue.id, changes: { priority } }]);
+      onUpdateIssues([
+        {
+          id: issue.id,
+          changes: { priority },
+          undoChanges: { priority: issue.priority },
```

Comment:
Ah, undoing field updates is an interesting bit I didn't fully consider. We have to get the field back to the value it was when the action happened.

The way you have it here looks correct, but I don't like the way that at every call site the caller has to remember to pass in the correct previous field and value. It would be easy to accidentally copy/paste the wrong undoChanges.

Can we either (a) have the caller pass the entire original value, then the handler gets the prev field values from that, or (b) have the handler look up the previous value from replicache and get the prev field values from there.

I know that in a previous code review there was something similar to this and I said it was weird to look up the entity from Replicache since caller already has it. I can't remember the details of that case but if you can find it we can compare and contrast.

Reply:
I believe the original way I did this was (b) but you recommended we use (a). I'll try to implement (a)